core/src/main/scala/kafka/log/Log.scala

```diff
@@ -217,6 +218,9 @@ class Log(@volatile var dir: File,
         if (indexFileExists) {
           try {
             segment.index.sanityCheck()
+            // Resize the time index file to 0 if it is newly created.
+            if (!timeIndexFileExists)
+              segment.timeIndex.resize(0)
```

Comment:
Hey Becket, can you point me to the code that will cause all .timeindex files (except for the one on the active segment) to be trimmed to 0 size upon broker restart? I am probably wrong, but it seems that broker always set file size to be config.maxIndexSize for both log index and time index.

Besides, if we were to set size of newly created time index file to 0, should we also do it in LogSegment.recover()?